### PR TITLE
Use cabalProject from haskell.nix

### PR DESCRIPTION
**Pull request by hamishmack**

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -1,3 +1,5 @@
+index-state: 2019-11-06T00:00:00Z
```
Comment (erikd) [inline cabal.project:1]: I have no problem with something like this `index-state` as an build output but am much less keen on it as an input.

Either that or have a `cabal` command to update that date to the date of the last Hackage index the user has pulled down if the local one is newer.

File: cabal.project [hunk 2]
```diff
@@ -54,166 +56,206 @@ package typed-protocols-cbor
...
+  tests: False
+
+package typed-protocols
+  tests: False
+
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-crypto
   tag: 4590efa638397e952a51a8994b5543e4ea3c1ecd
+  --sha256: 0hl2n3bba5v2j0lmxhs7hs01z3aznh2bwf9cb434icq2g0bl8ms3
```
Comment (erikd) [inline cabal.project:75]: Firstly I think that should have a better name that `sha256`. I would suggest `nix-sha256` to make it obvious what it is.

Secondly, (and this is *really* important) if I do not use Nix locally, how do I update these hashes?

File: cabal.project (hunk 1 above)
Comment (angerman) [reply cabal.project:1]: This is going to be a bit tricky. For reproducability, we need to ensure the solver will always produce the same solution. We can do this only with an index-state. haskell.nix updates the index-state once per day. So ideally you don't want to update this outside of the YYYY-MM-DD part.

Having cabal update this automatically give me ulcurs.

File: cabal.project (hunk 2 above)
Comment (angerman) [reply cabal.project:75]:
```
$ ssh user@some-nix-host nix-prefetch-git --quiet https://github.com/input-output-hk/cardano-crypto 4590efa638397e952a51a8994b5543e4ea3c1ecd |jq -r .sha256
0hl2n3bba5v2j0lmxhs7hs01z3aznh2bwf9cb434icq2g0bl8ms3
```

Maybe we can automate this with a regexp over the `cabal.project`?

**Review by angerman — COMMENTED** [review]
Alright, let's see how much we can fix from this. I'm afraid we won't be able to make everything be invisible.

**Comment by angerman** [comment]
replaced by #224

**Pull request closed without merging**